src/c++/perf_analyzer/genai-pa/genai_pa/main.py

```diff
@@ -25,24 +25,60 @@
 # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+import contextlib
+import io
 import logging
 import sys
 
 from genai_pa import parser
 from genai_pa.constants import LOGGER_NAME
 from genai_pa.exceptions import GenAiPAException
+from genai_pa.llm_inputs.llm_inputs import LlmInputs
+
+# Silence tokenizer warning on import
+with contextlib.redirect_stdout(io.StringIO()) as stdout, contextlib.redirect_stderr(
+    io.StringIO()
+) as stderr:
+    from genai_pa.llm_metrics import LLMProfileData
+    from transformers import AutoTokenizer as tokenizer
+    from transformers import logging as token_logger
+
+    token_logger.set_verbosity_error()
+
 
 logging.basicConfig(level=logging.INFO, format="%(name)s - %(levelname)s - %(message)s")
 logger = logging.getLogger(LOGGER_NAME)
 
 
+def generate_inputs(args):
+    LlmInputs.create_openai_llm_inputs(
+        args.dataset,
+        LlmInputs.DEFAULT_STARTING_INDEX,
+        LlmInputs.DEFAULT_LENGTH,
+        args.model,
+        args.streaming,
+    )
+
+
+def calculate_metrics(file: str) -> LLMProfileData:
+    t = tokenizer.from_pretrained("gpt2")
+    return LLMProfileData(file, t)
+
+
+def report_output(metrics: LLMProfileData):
+    print(metrics.get_statistics("concurrency", 1))
```

Comment:
I don't totally get this one, why do we query a specific scenario here (concurrency==1) rather than walking through all the results/scenarios and printing them?

Reply:
or querying whatever was passed to the CLI whether `--concurrency` or `--request-rate` etc

Reply:
you are correct.
This was trying to hook things up quickly to ensure that everything was working. 
I will update this to be more appropriate.